### PR TITLE
openssl_privatekey_info: move main code to module_utils to allow easier implementation of diff mode

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Subset of #150.

##### ISSUE TYPE
- Refactoring Pull Request

##### COMPONENT NAME
openssl_privatekey_info

### CODE REVIEW

File: plugins/modules/openssl_privatekey_info.py [hunk 1]
```diff
@@ -601,49 +170,39 @@ def main():
...
-                backend = 'cryptography'
-            elif can_use_pyopenssl:
-                backend = 'pyopenssl'
+    if module.params['content'] is not None:
+        data = module.params['content'].encode('utf-8')
+    else:
+        try:
+            with open(module.params['path'], 'rb') as f:
+                data = f.read()
+        except Exception as e:
```
Comment (Ajpantuso) [inline plugins/modules/openssl_privatekey_info.py:185]:
```suggestion
        except (IOError, OSError) as e:
```

File: plugins/module_utils/crypto/module_backends/privatekey_info.py [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+        return backend, PrivateKeyInfoRetrievalPyOpenSSL(
+            module, content, passphrase=passphrase, return_private_key_data=return_private_key_data)
+    elif backend == 'cryptography':
+        if not CRYPTOGRAPHY_FOUND:
+            module.fail_json(msg=missing_required_lib('cryptography >= {0}'.format(MINIMAL_CRYPTOGRAPHY_VERSION)),
+                             exception=CRYPTOGRAPHY_IMP_ERR)
+        return backend, PrivateKeyInfoRetrievalCryptography(
+            module, content, passphrase=passphrase, return_private_key_data=return_private_key_data)
+    else:
+        raise Exception('Unsupported value for backend: {0}'.format(backend))
```
Comment (Ajpantuso) [inline plugins/module_utils/crypto/module_backends/privatekey_info.py:490]:
```suggestion
        raise ValueError('Unsupported value for backend: {0}'.format(backend))
```

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
No functional issues and the new exceptions raised in `get_info` match the previous "direct" calls to `fail_json`

**Review by Ajpantuso — APPROVED** [review]
LGTM, thanks!

**Comment by felixfontein** [comment]
@Ajpantuso thanks again for reviewing this one as well! :)

**Pull request merged**